server/methods/cart.js

```diff
@@ -0,0 +1,149 @@
+        return label.vendor + label.title;
+      }
+      selectedOptions.push(bundleSelection);
+    });
+
+    ReactionCore.Collections.Cart.update({
+      '_id': cart._id,
+      'items.productId': productId,
+      'items.variants._id': variantId
+    }, {
+      $set: {
+        'items.$.variants.selectedBundleOptions': selectedOptions
+      }
+    });
+  },
+  'productBundler/addBundleItemToCart': function (productId, variantId, bundleId, bundleIndex) {
+    check(productId, String);
+    check(variantId, String);
+    check(bundleId, String)
+    check(bundleIndex, Number);
+    const { Log } = ReactionCore;
+    const cart = ReactionCore.Collections.Cart.findOne({ userId: this.userId });
+    if (!cart) {
+      Log.error(`Cart not found for user: ${ this.userId }`);
+      throw new Meteor.Error(404, "Cart not found",
+        "Cart not found for user with such id");
+    }
+    // With the flattened model we no longer need to work directly with the
+    // products. But product still could be necessary for a `quantityProcessing`
+    // TODO: need to understand: do we really need product inside
+    // `quantityProcessing`?
+    let product;
+    let variant;
+    ReactionCore.Collections.Products.find({ _id: { $in: [
+      productId,
+      variantId
+    ]}}).forEach(doc => {
+      if (doc.type === "simple") {
+        product = doc;
+      } else {
+        variant = doc;
+      }
+    });
+    // TODO: this lines still needed. We could uncomment them in future if
+    // decide to not completely remove product data from this method
+    // const product = ReactionCore.Collections.Products.findOne(productId);
+    // const variant = ReactionCore.Collections.Products.findOne(variantId);
+    if (!product) {
+      Log.warn(`Product: ${ productId } was not found in database`);
+      throw new Meteor.Error(404, "Product not found",
+        "Product with such id was not found!");
+    }
+    if (!variant) {
+      Log.warn(`Product variant: ${ variantId } was not found in database`);
+      throw new Meteor.Error(404, "ProductVariant not found",
+        "ProductVariant with such id was not found!");
+    }
+    let quantity = 1;
+
+    return ReactionCore.Collections.Cart.update({
+      _id: cart._id
+    }, {
+      $addToSet: {
+        items: {
+          _id: Random.id(),
+          shopId: product.shopId,
+          productId: productId,
+          quantity: quantity,
+          variants: variant,
+          type: product.type,
+          customerViewType: 'bundleComponent',
+          bundleProductId: bundleId,
+          bundleIndex: bundleIndex
+        }
+      }
+    }, function (error, result) {
+      if (error) {
+        Log.warn("error adding to cart", ReactionCore.Collections.Cart
+          .simpleSchema().namedContext().invalidKeys());
+        return error;
+      }
+
+      // refresh shipping quotes
+      // Meteor.call("shipping/updateShipmentQuotes", cart._id);
```

Comment:
Did we decide we don't need to updateShipmentQuotes? is this only customer facing?

Lets remove these comments / commented out method calls.